provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/action/BlackDuckDistributionTestAction.java

```diff
@@ -72,12 +74,15 @@ public MessageResult testConfig(String configId, FieldModel fieldModel, FieldUti
                 .flatMap(projectNamePattern -> validatePatternMatchesProject(providerConfigId, projectNamePattern))
                 .ifPresent(fieldStatuses::add);
 
-            Optional<BlackDuckProperties> optionalBlackDuckProperties = configurationAccessor.getConfigurationById(providerConfigId)
-                                                                            .map(blackDuckProvider::createStatefulProvider)
-                                                                            .map(statefulProvider -> (BlackDuckProperties) statefulProvider.getProperties());
-            if (optionalBlackDuckProperties.isPresent()) {
-                BlackDuckProperties blackDuckProperties = optionalBlackDuckProperties.get();
+            BlackDuckProperties blackDuckProperties = null;
+            Optional<ConfigurationModel> providerConfigurationOptional = configurationAccessor.getConfigurationById(providerConfigId);
+            if (providerConfigurationOptional.isPresent()) {
+                ConfigurationModel providerConfiguration = providerConfigurationOptional.get();
+                StatefulProvider statefulProvider = blackDuckProvider.createStatefulProvider(providerConfiguration);
+                blackDuckProperties = (BlackDuckProperties) statefulProvider.getProperties();
```

Comment:
Can this be changed to optional mapping? It would remove this if statement and you could orElse(null) at the end.

Reply:
`blackDuckProvider.createStatefulProvider(providerConfiguration)` throws an exception so we can't change this if block to use the Optional.map function.